### PR TITLE
foundation for locations, coords, location searching

### PR DESCRIPTION
**Pull request by mgeist**
we are gonna need to find a way to mock out the googlemaps client for unit tests

### CODE REVIEW

File: core/models/coords.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      .extra(
+        select={'distance': formula},
+        select_params=[lat, lon, lat],
+        where=[distance_lt],
+        params=[lat, lon, lat, proximity],
+        order_by=['distance']))
+
+
+class Coords(models.Model):
+  event = models.ForeignKey(Event, related_name="coords")
```
Comment (mgeist) [inline core/models/coords.py]: make this 1:1

Comment (vivianliang) [inline core/models/coords.py]: what happens if 2 events are at the same place?

Comment (mgeist) [inline core/models/coords.py]: theyll each just have a coords object, the lat/lon fields wont be unique

Comment (mgeist) [inline core/models/coords.py]: but conversely we need to consider an event being able to be at 2 places..

**Pull request merged**